src/diffpy/labpdfproc/functions.py

```diff
@@ -182,9 +172,62 @@ def get_path_length(self, grid_point, angle):
         return total_distance, primary_distance, secondary_distance
 
 
+def _cve_brute_force(diffraction_data, mud, wavelength):
+    """
+    compute cve using brute-force method
+
+    it is computed as follows:
+    We first resample data and absorption correction to a more reasonable grid,
+    then calculate corresponding cve for the given mud in the resample grid
+    (since the same mu*D yields the same cve, we can assume that D/2=1, so mu=mud/2),
+    and finally interpolate cve to the original grid in diffraction_data.
+    """
+
+    mu_sample_invmm = mud / 2
+    abs_correction = Gridded_circle(mu=mu_sample_invmm)
+    distances, muls = [], []
+    for angle in TTH_GRID:
+        abs_correction.set_distances_at_angle(angle)
+        abs_correction.set_muls_at_angle(angle)
+        distances.append(sum(abs_correction.distances))
+        muls.append(sum(abs_correction.muls))
+    distances = np.array(distances) / abs_correction.total_points_in_grid
+    muls = np.array(muls) / abs_correction.total_points_in_grid
+    cve = 1 / muls
+    return cve
+
+
+def _cve_interp_polynomial(diffraction_data, mud, wavelength):
+    """
+    compute cve using polynomial interpolation method, raise an error if mu*D is out of the range (0.5 to 6)
+    """
+
+    if mud > 6 or mud < 0.5:
+        raise ValueError(
+            "mu*D is out of the acceptable range (0.5 to 6) for fast calculation. "
+            "Please rerun with a value within this range or use -b to enable brute-force calculation. "
+        )
+    coeff_a, coeff_b, coeff_c, coeff_d, coeff_e = [
+        interpolation_function(mud) for interpolation_function in INTERPOLATION_FUNCTIONS
+    ]
+    muls = np.array(coeff_a * MULS**4 + coeff_b * MULS**3 + coeff_c * MULS**2 + coeff_d * MULS + coeff_e)
+    cve = 1 / muls
+    return cve
+
+
+def _cve_method(diffraction_data, mud, wavelength, brute_force=False):
```

Comment:
this api will break if we extend the methods.  Better is to pass in method and give it a name, so `method="brute-force"` and so on.

For the argparser maintain a global list `CVE_METHODS=["brute_force", "polynomial_interpolation"]` in the top of the app main and then list these in the argparser help, so if someone types --help they can see the valid values

If we add a new method we just need to add it in three places, but the code API won't change.

Reply:
Yes, sorry, I got this now. I'll correct this in the next commit.